tests/pytorch/mock_sparse/test_reduce.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import torch
+import numpy
+
+from dgl.mock_sparse import create_from_coo
+
+
+@pytest.mark.parametrize("dense_dim", [None, 2])
+@pytest.mark.parametrize("row", [[0, 0, 1, 2], (0, 1, 2, 4)])
+@pytest.mark.parametrize("col", [(0, 1, 2, 2), (1, 3, 3, 4)])
+@pytest.mark.parametrize("mat_shape", [(3, 5), (5, 3)])
+@pytest.mark.parametrize("reduce_type", ['sum', 'smax', 'smin', 'mean'])
+@pytest.mark.parametrize("dim", [None, 0, 1])
+def test_reduce(dense_dim, row, col, mat_shape, reduce_type, dim):
+    # Skip invalid matrices
+    if max(row) >= mat_shape[0] or max(col) >= mat_shape[1]:
```

Comment:
Perhaps it's better to replace `mat_shape` with `extra_shape` or `shape_offset` for expanding the shape dictated with `row` and `col`.

Reply:
I didn't get your point. Basically I think mat_shape cannot be inferred from `row` and `col`.

Reply:
Basically, we have

```
base_shape = (max(row) + 1, max(col) + 1)
target_shape = base_shape + offset_shape
```

Reply:
I see. I will change them accordingly.

Reply:
Done.